pvlib/iotools/solcast.py

```diff
@@ -0,0 +1,459 @@
+    metadata: dict
+        latitude and longitude of the request.
+
+    Examples
+    --------
+    >>> df, meta = pvlib.iotools.solcast.get_solcast_historic(
+    >>>     latitude=-33.856784,
+    >>>     longitude=151.215297,
+    >>>     start='2007-01-01T00:00Z',
+    >>>     duration='P1D',
+    >>>     api_key="your-key"
+    >>> )
+
+    you can pass any of the parameters listed in the API docs,
+    for example using the ``end`` parameter instead
+
+    >>> df, meta = pvlib.iotools.solcast.get_solcast_historic(
+    >>>     latitude=-33.856784,
+    >>>     longitude=151.215297,
+    >>>     start='2007-01-01T00:00Z',
+    >>>     end='2007-01-02T00:00Z',
+    >>>     api_key="your-key"
+    >>> )
+
+    References
+    ----------
+    .. [1] `Get an API Key <https://toolkit.solcast.com.au/register>`_
+    .. [2] `Solcast API Docs <https://docs.solcast.com.au/>`_
+    """
+
+    params = dict(
+        latitude=latitude,
+        longitude=longitude,
+        start=start,
+        end=end,
+        duration=duration,
+        api_key=api_key,
+        format="json",
+        **kwargs
+    )
+
+    data = _get_solcast(
+        endpoint="historic/radiation_and_weather",
+        params=params,
+        api_key=api_key,
+        map_variables=map_variables
+    )
+
+    return data, {"latitude": latitude, "longitude": longitude}
+
+
+def get_solcast_forecast(
+    latitude, longitude, api_key, map_variables=True, **kwargs
+):
+    """Get irradiance and weather forecasts from the present time
+    up to 14 days ahead
+
+    Parameters
+    ----------
+    latitude : float
+        in decimal degrees, between -90 and 90, north is positive
+    longitude : float
+        in decimal degrees, between -180 and 180, east is positive
+    api_key : str
+        To access Solcast data you will need an API key [1]_.
+    map_variables: bool, default: True
+        When true, renames columns of the DataFrame to pvlib variable names
+        where applicable. See variable :const:`VARIABLE_MAP`.
+    kwargs:
+        Optional parameters passed to the API.
+        See [2]_ for full list of parameters.
+
+    Returns
+    -------
+    data : pandas.DataFrame
+        Contains the values for the parameters requested.The times
+        in the DataFrame index indicate the midpoint of each interval.
+    metadata: dict
+        latitude and longitude of the request.
+
+    Examples
+    --------
+    >>> df, meta = pvlib.iotools.solcast.get_solcast_forecast(
+    >>>     latitude=-33.856784,
+    >>>     longitude=151.215297,
+    >>>     api_key="your-key"
+    >>> )
+
+    you can pass any of the parameters listed in the API docs,
+    like asking for specific variables:
+
+    >>> df, meta = pvlib.iotools.solcast.get_solcast_forecast(
+    >>>     latitude=-33.856784,
+    >>>     longitude=151.215297,
+    >>>     output_parameters=['dni', 'clearsky_dni', 'snow_soiling_rooftop'],
+    >>>     api_key="your-key"
+    >>> )
+

```

Comment:
I think it would be nice to add forecast horizon to this example using the ``hours`` parameter.